added apoc stub

wjurkowski opened the pull request.

wjurkowski committed:
added apoc stub

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,280 @@
+              title = {{APoc}: large-scale identification of similar protein pockets},
+              journal = {Bioinformatics}
+            }
+        </citation>
```
`<citation type="doi">10.1093/bioinformatics/btt024</citation>` should be sufficient

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,280 @@
+<tool id="apoc" name="APoc" version="1.0">
+    <description>Large-scale identification of similar protein pockets</description>
+    <requirements>
+        <requirement type="package" version="1b16">acpc</requirement>
```
acpc -> apoc

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,280 @@
+    </stdio>
+    <command>
+        <![CDATA[ 
+        #if $templates_source.template_source_select=="list"#
```
You don't normally need the `#` at the end of a Cheetah directive.

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,280 @@
+                                                -lt templates_list
+                                            #else
+                                                #if $templates_source.pt
+                                                    -pt $templates_source.pt
```
Single quotes `'` around `$templates_source.pt` (it's of type `text`, so can contain spaces and other nasty characters).

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,280 @@
+        <option value="1" selected="true">enable </option>
+        <option value="0">disable </option>
+        </param>
+        <param name="pvol" type="text" value="1000" label="Minimal pocket volume in grid points" help="(-pvol)" />
```
`type="integer"`

nsoranzo commented on tools/apoc/test-data/apoc/block.pdb:
```diff
@@ -0,0 +1,8087 @@
+PDBSTA 1yr8A.pdb
```
This file doesn't seem to be used.

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,280 @@
+        <option value="0">disable </option>
+        </param>
+        <param name="pvol" type="text" value="1000" label="Minimal pocket volume in grid points" help="(-pvol)" />
+        <param name="plen" type="text" value="10" label="Minimal number of pocket residues" help="(-plen)" />
```
`type="integer"`

wjurkowski committed:
apoc citation fix

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+                </repeat>
+            </when>
+        </conditional>
+        <param name="fa" type="select" label="Global structure alignment" help="(-fa)">
```
You can use the `argument` attribute instead of specifying the option of the underlying tool inside `help`. When the `name` attribute is derived from it by just removing the initial dashes, you can also omit it. So that could be rewritten as:
```xml
<param argument="-fa" type="select" label="Global structure alignment">
```

The same applies for all `<param>`s below.

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+        </conditional>
+        <param name="fa" type="select" label="Global structure alignment" help="(-fa)">
+        <option value="1" selected="true">enable </option>
+        <option value="0">disable </option>
```
The 2 lines above are not correctly indented.

nsoranzo reviewed (commented):
Thanks @wjurkowski!

nsoranzo commented:
I've created a BioConda recipe for apoc in https://github.com/bioconda/bioconda-recipes/pull/5114 .

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+            paste -d "\t" queries_path $query_source.lq | awk '{print $1"\t"$3}' > queries_list;
+        #end if#
+
+        $__tool_directory__/tools/apoc/apoc
```
This should just be `apoc`

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+                                                #if $query_source.pq
+                                                    -pq $query_source.pq
+                                                #end if 
+                                            $query_source.pdbfile2
```
Indentation

wjurkowski replied:
Thanks @nsoranzo!!!
Working on it

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+                                            -pvol $pvol
+                                            -plen $plen
+                                            #if $alignment_option.sod=="true"
+                                            -sod
```
Indentation

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+
+                                            -m $scoring_option.m
+                                            #if $scoring_option.L
+                                            -L $scoring_option.L
```
Indentation

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+                                            -L $scoring_option.L
+                                            #end if
+                                            #if $scoring_option.a=="true"
+                                            -a
```
Indentation

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+                                            -a
+                                            #end if
+                                            #if $scoring_option.b=="true"
+                                            -b
```
Indentation

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,266 @@
+                                            -b
+                                            #end if
+                                            #if $scoring_option.c=="true"
+                                            -c
```
Indentation

wjurkowski committed:
fixes in apoc.xml

wjurkowski committed:
removed block.pdb

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+    </stdio>
+    <command>
+        <![CDATA[ 
+        #if '$templates_source.template_source_select'=="list"#
```
You shouldn't use quotes around variables in the Cheetah directives (lines starting with `#`), only in the lines that end up in the job script (and for variables corresponding to `<param>s` of type `text`, `data`, `data_collection` or similar).

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+            paste -d "\t" queries_path $query_source.lq | awk '{print $1"\t"$3}' > queries_list;
+        #end if#
+
+        $__tool_directory__/tools/apoc/
```
I really meant just `apoc`! When Galaxy resolves the requirement installing the BioConda package, `apoc` will be on the `$PATH`.

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+            paste -d "\t" templates_path '$templates_source.lt' | awk '{print $1"\t"$3}' > templates_list;
+        #end if#
+
+        #if '$query_source.query_source_select'=="list"#
```
As above, don't quote `$query_source.query_source_select`

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+                                                -c
+                                            #end if
+
+                                            #if '$templates_source.template_source_select'=="list"
```
As above, don't quote `$templates_source.template_source_select`

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+                                            #if '$templates_source.template_source_select'=="list"
+                                                -lt templates_list
+                                            #else
+                                                #if '$templates_source.pt'
```
Remove quotes

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+                                                '$templates_source.pdbfile1'
+                                            #end if
+
+                                            #if '$query_source.query_source_select'=="list"
```
As above, remove quotes around `$query_source.query_source_select`

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+                                            #if '$query_source.query_source_select'=="list"
+                                                -lq queries_list
+                                            #else
+                                                #if '$query_source.pq'
```
Remove quotes

wjurkowski committed:
fixes in apoc.xml

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+                                            #if $query_source.query_source_select=="list"
+                                                -lq queries_list
+                                            #else
+                                                #if '$query_source.pq'
```
Remove quotes

nsoranzo commented on tools/apoc/apoc.xml:
```diff
@@ -0,0 +1,265 @@
+                                                #if $templates_source.pt
+                                                    -pt '$templates_source.pt'
+                                                #end if 
+                                                $templates_source.pdbfile1
```
Re-add quotes :D

nsoranzo reviewed (commented):
We're almost there, 2 tests out of 4 pass now!

wjurkowski committed:
more edits of apoc.xml

wjurkowski commented:
Thanks!

nsoranzo committed:
Final cleanup of apoc

- Use boolean type for `-fa`, `-sod`, `-a`, `-b` and `-c` params
- Use integer type for `-L` param
- Remove redundant help
- Use `detect_errors="exit_code"` instead of `<stdio>`

nsoranzo commented:
@wjurkowski All tests pass now! I've added a final cleanup in commit https://github.com/TGAC/earlham-galaxytools/pull/88/commits/8d552f7c955ed8b200a88462218788eefdac49c9 , if that looks good to you I'll merge.

wjurkowski commented:
@nsoranzo thanks for last checks and fixes! All looks good to me
